core/src/OpenMP/Kokkos_OpenMP_Instance.cpp

```diff
@@ -62,6 +62,8 @@ void OpenMPInternal::clear_thread_data() {
     if (nullptr != m_pool[rank]) {
       m_pool[rank]->disband_pool();
 
+      m_pool[rank]->~HostThreadTeamData();
```

Comment:
This is to match the placement new at https://github.com/kokkos/kokkos/blob/a1b47dbb6687c41f4ca1ffa3123178741d6d6554/core/src/OpenMP/Kokkos_OpenMP_Instance.cpp#L130.